File: VehicleViolationParking/sql/import_db.py
import pymysql
import sys
import os

db_config = {
    'host': '127.0.0.1',
    'port': 3306,
    'user': 'root',
    'password': '123456',
    'charset': 'utf8mb4'
}

sql_file_path = os.path.join(os.path.dirname(os.path.abspath(__file__)), 'vehicle_parking.sql')

try:
    conn = pymysql.connect(**db_config)
    cursor = conn.cursor()

    cursor.execute("CREATE DATABASE IF NOT EXISTS `vehicle_parking` DEFAULT CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci")
    conn.commit()
    print("Database vehicle_parking created/verified.")

    cursor.execute("USE `vehicle_parking`")

    with open(sql_file_path, 'r', encoding='utf-8') as f:
        sql_content = f.read()

    statements = sql_content.split(';')
    for stmt in statements:
        stmt = stmt.strip()
        if not stmt or stmt.startswith('--') or 'CREATE DATABASE' in stmt.upper() or 'USE `' in stmt:
            continue
        try:
            cursor.execute(stmt)
            conn.commit()
        except Exception as e:
            print(f"Warning - SQL error: {e}")
            print(f"Statement: {stmt[:120]}...")
            conn.rollback()

    cursor.execute("SHOW TABLES FROM vehicle_parking")
    tables = cursor.fetchall()
    print("\nSuccessfully imported. Tables:")
    for table in tables:
        print(f"  - {table[0]}")

    print("\nData counts:")
    for table in tables:
        tname = table[0]
        try:
            cursor.execute(f"SELECT COUNT(*) FROM `{tname}`")
            count = cursor.fetchone()[0]
            print(f"  {tname}: {count} rows")
        except:
            pass

    cursor.close()
    conn.close()
    print("\nDone.")

except Exception as e:
    print(f"Error: {e}")
    import traceback
    traceback.print_exc()
    sys.exit(1)

File: LearningDiagnosisKnowledge/database/import-db.js
const mysql = require('mysql2/promise');
const fs = require('fs');
const path = require('path');

async function importDatabase() {
  console.log('开始连接 MySQL 数据库...');
  
  const connection = await mysql.createConnection({
    host: '127.0.0.1',
    port: 3306,
    user: 'root',
    password: '123456',
    multipleStatements: true
  });

  console.log('数据库连接成功!');

  try {
    const sqlPath = path.join(__dirname, 'schema.sql');
    let sqlContent = fs.readFileSync(sqlPath, 'utf8');
    
    console.log('开始执行 SQL 脚本...');
    await connection.query(sqlContent);
    console.log('✅ 数据库脚本导入成功!');
    
    const [rows] = await connection.query('SHOW TABLES FROM learning_diagnosis');
    console.log(`\n📊 已创建 ${rows.length} 张表:`);
    rows.forEach(row => {
      const tableName = Object.values(row)[0];
      console.log(`   - ${tableName}`);
    });

    const [counts] = await connection.query(`
      SELECT 
        (SELECT COUNT(*) FROM learning_diagnosis.users) as users,
        (SELECT COUNT(*) FROM learning_diagnosis.subjects) as subjects,
        (SELECT COUNT(*) FROM learning_diagnosis.knowledge_points) as kps,
        (SELECT COUNT(*) FROM learning_diagnosis.questions) as questions,
        (SELECT COUNT(*) FROM learning_diagnosis.classes) as classes
    `);
    
    console.log('\n📈 初始化数据:');
    console.log(`   - 用户: ${counts[0].users} 人`);
    console.log(`   - 学科: ${counts[0].subjects} 个`);
    console.log(`   - 知识点: ${counts[0].kps} 个`);
    console.log(`   - 题目: ${counts[0].questions} 道`);
    console.log(`   - 班级: ${counts[0].classes} 个`);

  } catch (error) {
    console.error('❌ 导入失败:', error.message);
    process.exit(1);
  } finally {
    await connection.end();
    console.log('\n✅ 数据库连接已关闭');
  }
}

importDatabase();
